### ISSUE TITLE
Server: errors relating to MD057 can't be cleared without additional text entry

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#548]
### rumdl version

0.1.61

### Installation method

mise

### Bug description

This bug is closely related to: https://github.com/rvben/rumdl/issues/517, and might be the same. After https://github.com/rvben/rumdl/issues/517 was closed, I could still reproduce it (or something like it): https://github.com/rvben/rumdl/issues/517#issuecomment-4052064248.

Basically, I've narrowed the problem down to needing to provide text input to the editor after fixing a violation of rule MD057.

### Steps to reproduce

1. Create two Markdown file:
   - First Markdown file:
     ```markdown
     ---
     title: Heading
     ---
     
     Here is [a link](linked.md)
     ```
   - Second Markdown file (called `linked.md`) in the same directory as the first file:
     ```markdown
     ---
     title: Heading
     ---
     
     Here's some Markdown.
     ```
1. Update the first Markdown file in the editor to be:
   ```markdown
   ---
   title: Heading
   ---
   
   Here is [a link](../linked.md)
   ```
   The editor will show an error that the link is invalid.
1. Restore the first Markdown file to its previous state. The error is still present in the editor.
1. In the first Markdown file, add a full stop to the end of line `5`. The error is cleared.

### Expected behavior

Given the scenario above, I'd expect `rumdl server` to work such that as soon as a keystroke fixes the error, the editor shows no error.

### Actual behavior

Given the scenario above, I need to provide additional text input to the editor before the editor shows no error.

### Configuration

None.

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.62. The [MD057](https://rumdl.dev/md057) file existence cache is now reset at the start of `cross_file_check()`, preventing stale diagnostics from persisting after fixing a broken link. Thanks for reporting, @eread!
